geniza/corpus/management/commands/import_data.py

```diff
@@ -142,19 +144,26 @@ def import_collections(self):
     def import_languages(self):
         LanguageScript.objects.all().delete()
         language_data = self.get_csv('languages')
-        languages = LanguageScript.objects.bulk_create([
-            LanguageScript(language=row.language,
-                           script=row.script,
-                           display_name=row.display_name)
-            for row in language_data if row.language and row.script
-        ])
+        languages = []
+        for row in language_data:
+            # skip empty rows
+            if not row.language and not row.script:
+                continue
+
+            lang = LanguageScript.objects.create(
+                language=row.language,
+                script=row.script,
+                display_name=row.display_name or None)
+
+            # populate lookup for associating documents & languages;
+            # use lower case spreadsheet name if set, or display name
+            if row.display_name or row.spreadsheet_name:
+                self.language_lookup[(row.spreadsheet_name or
+                                      row.display_name).lower()] = lang
```

Comment:
oooo okay. Yeah this gives me a bit of a headache, but I agree this is the best way to handle it 🥴